### ISSUE TITLE
Case sensitivity of parameters. If capitol the parser fails to recognize it.

### ISSUE THREAD

**Issue by wsmorgan** [rosenbrockc/fortpy#44]
code:
CONTAINS
  !!<summary>This routine takes an integer 3x3 matrix and computes its
  !!Smith Normal Form.</summary>
  !!<parameter name="H" regular="True">Input matrix.</parameter>
  !!<parameter name="A" regular="True">Smith Normal Form.</parameter>
  !!<parameter name="M" regular="True">Left Transform.</parameter>
  !!<parameter name="B" regular="True">Right Transform.</parameter>
  subroutine SmithNormalForm(H,A,M,B)
error:
Traceback (most recent call last):
  File "/Users/wileymorgan/.virtualenvs/fortpy/bin/runtests.py", line 91, in <module>
    initialize()
  File "/Users/wileymorgan/.virtualenvs/fortpy/bin/runtests.py", line 39, in initialize
    t.writeall(args["codedir"])
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/tester.py", line 503, in writeall
    self.tgenerator.write(self._codefolder)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/generator.py", line 86, in write
    self._write_module(module)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/generator.py", line 102, in _write_module
    self._write_executable(module, anexec)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/generator.py", line 197, in _write_executable
    self.xgenerator.write(testid)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/executable.py", line 146, in write
    lines.append(self.writer.lines(testid))
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/method.py", line 139, in lines
    self.method.group.code(testid, "after", result, "  ")
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/elements.py", line 1984, in code
    self._codes[testid][section](lines, spacer)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/elements.py", line 1717, in code_after
    t.code(lines, self.variables, "each", spacer)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/elements.py", line 1440, in code
    self._process(variables, spacer)
  File "/Users/wileymorgan/.virtualenvs/fortpy/lib/python2.7/site-packages/fortpy/testing/elements.py", line 1492, in _process
    "a <global> tag or regular=true parameter doc tag.")
ValueError: Variable m has not been initialized with a <global> tag or regular=true parameter doc tag.

**Comment by wsmorgan** [comment]
Add `.lower()` to the XML attribute test for equality on line #2068 in `elements.py`:

```python
"regular" in doc.attributes and doc.attributes["regular"].lower() == "true":
```

**Comment by rosenbrockc** [comment]
Fixed in Revision 1.5.7 by introducing case insensitivity to the `regular="true"` parameter check and by fixing a new bug we discovered where if `dimension` was added as a _modifier_ instead of attached to the variable name, it wasn't being parsed correctly by `parsers.variable`.
